packages/hurl/src/http/client.rs

```diff
@@ -488,7 +488,12 @@ impl Client {
         self.set_multipart(&request_spec.multipart)?;
         let request_spec_body = &request_spec.body.bytes();
         self.set_body(request_spec_body)?;
-        let headers_spec = &request_spec.headers;
+        let options_headers = options
+            .headers
+            .iter()
+            .map(|h| h.as_str())
+            .collect::<Vec<&str>>();
+        let headers_spec = &request_spec.headers.aggregate_raw_headers(&options_headers);
```

Comment:
What we could change here is the name of the variable and the parameter : it's not `let headers_spec`it's now `let headers`. A tiny change just to reiterate that now `headers` is not from the spec only, it's and aggregation of the spec and the command line.

So we have now:

```rust
let headers = &request_spec.headers.aggregate_raw_headers(&options_headers);
```

and

```rust
    fn set_headers(
        &mut self,
        headers: &HeaderVec,
        implicit_content_type: Option<&str>,
        options: &ClientOptions,
    ) -> Result<(), HttpError> {
```

I like that the method is just `fn set_headers(..., headers: &HeaderVec...)`☺️